app/models/product_test.rb

```diff
@@ -104,6 +103,23 @@ def archive_records
     save
   end
 
+  def sample_and_duplicate_records(recs, ids, random: Random.new)
+    car = ::Validators::CalculatingAugmentedRecords.new(measures, [], id)
+    dups = records.find(ids)
+
+    (random.rand(3) + 1).times do
+      prng_repeat = Random.new(rand_seed.to_i)
+      dup_rec, rec_augments, old_rec = dups.sample(random: random).duplicate_randomization(random: prng_repeat)
+      if car.validate_calculated_results(dup_rec, 'effective_date' => effective_date)
+        augmented_records << rec_augments
+        recs << dup_rec
+      else
+        recs << old_rec
```

Comment:
Should something be added to "augmented_records" to indicate that the patient is "duplicate" but no other changes have been made?  Maybe just at the record_id?